cli/internal/state/state.go

```diff
@@ -30,9 +35,99 @@ func ReadFromFile(fileHandler file.Handler, path string) (*State, error) {
 
 // State describe the entire state to describe a Constellation cluster.
 type State struct {
-	Version        string         `yaml:"version"`
+	// description: |
+	//   Schema version of this state file.
+	Version string `yaml:"version"`
+
+	// TODO(msanft): Add link to self-managed infrastructure docs once existing.
+
+	// description: |
+	//   State of the cluster's cloud resources. These values are retrieved during
+	//   cluster creation. In the case of self-managed infrastructure, the marked
+	//   fields in this struct should be filled by the user as per
+	//   https://docs.edgeless.systems/constellation/workflows/create.
 	Infrastructure Infrastructure `yaml:"infrastructure"`
-	ClusterValues  ClusterValues  `yaml:"clusterValues"`
+	// description: |
+	//   DO NOT EDIT. State of the Constellation Kubernetes cluster.
+	//   These values are set during cluster initialization and should not be changed.
+	ClusterValues ClusterValues `yaml:"clusterValues"`
+}
+
+// ClusterValues describe the (Kubernetes) cluster state, set during initialization of the cluster.
+type ClusterValues struct {
+	// description: |
+	//   Unique identifier of the cluster.
+	ClusterID string `yaml:"clusterID"`
+	// description: |
+	//   Unique identifier of the owner of the cluster.
+	OwnerID string `yaml:"ownerID"`
+	// description: |
+	//   Salt used to generate the ClusterID on the bootstrapping node.
+	MeasurementSalt []byte `yaml:"measurementSalt"`
+}
+
+// Infrastructure describe the state related to the cloud resources of the cluster.
+type Infrastructure struct {
+	// description: |
+	//   Unique identifier the cluster's cloud resources are tagged with.
+	UID string `yaml:"uid"`
```

Comment:
should we indicate optional values with a pointer?

Reply:
This makes it quite tedious to work with in code, as you need non-nil checks everywhere. I see a value in using a pointer to the nested structs, but I wouldn't use them for all fields